guardian/grid
# imageLoader API endpoint for uploads by user

## What does this change?

Adds a new API endpoint to the ImageLoader to return the UploadStatus of all images by a given user (the current user if no userId is specified on the path).
Adds a methods to the loader api service in Kahuna to fetch the uploads data for the current user

## How should a reviewer test this change?

run locally:
 - open the kahuna UI to authenticate your browser session
 - open the api page at `https://loader.media.local.dev-gutools.co.uk/uploadStatuses`
 - Upload some (large) images from the UI
 - The statuses of your new images should be update on refreshing the api page as the uploads progress


## How can success be measured?
The new endpoint isn't in use yet - this is preparatory work for making improvements to the user experience for image uploads.

Currently, Kahuna polls the image API for each image uploaded (in the present browser session) to update the UI when the ingest is completed. This endpoint could be used instead to poll the loader API to see the progress of all the user's  uploads - even if the page was refreshed after the image was uploaded to the queue bucket.

the endpoint could also act as a quick diagnostic tool - if a particular user reports issues uploading images, we can use it to see the status of their uploads, including any failures.

## Who should look at this?


## Tested? Documented?
